classifier_and_summarizer/summarizer/text_media_matcher.py

```diff
@@ -0,0 +1,173 @@
+''' Text Media Matcher
+This module contains utilities for matching
+text and media .It uses the gale shapley algorithm
+for finding the stable matching - and it is optimal
+for the preference of the media.
+This script contains the following classes:
+    *ElementWithIndex : represent the text along with index
+    *TextMediaMatcher : implements the main algorithm for matching
+                        returns the stable matching
+'''
+from scipy.spatial.distance import cosine as cosine_distance
+from sentence_transformers import SentenceTransformer
+
+
+class ElementWithIndex:
```

Comment:
This will only be used here! In the output from the extractor it'll consist of the content along with index - so we find the media description and pair it with the index so it can be used for matching. Let me know if there is a better approach for this!

Reply:
This is fine in that case. Thanks!